test/groups/groups_general.jl

```diff
@@ -29,8 +24,8 @@ end
         @test_throws ErrorException compose(G, x, x)
         @test compose(G, eg, x) == x
         @test compose(G, x, eg) == x
-        @test compose!(G, x, eg, x) == x
-        @test compose!(G, x, x, eg) == x
+        @test_throws ErrorException compose!(G, x, eg, x)
```

Comment:
Why is an error expected here?

Reply:
There are no longer any defaults for `compose!` for identity elements. This is because once one overloads `compose!` for a new operation, one needs to reimplement these special cases to resolve ambiguity, so the general defaults can never be used.